src/Storages/MergeTree/MergeTreeSelectProcessor.cpp

```diff
@@ -15,6 +15,39 @@
 #include <Storages/MergeTree/MergeTreeVirtualColumns.h>
 #include <city.h>
 
+namespace
+{
+
+template <typename Func>
+struct TelemetryWrapper
+{
+    TelemetryWrapper(Func callback_, ProfileEvents::Event event_, std::string span_name_)
+        : callback(std::move(callback_)), event(event_), span_name(std::move(span_name_))
+    {
+    }
+
+    template <typename... Args>
+    auto operator()(Args &&... args)
+    {
+        DB::OpenTelemetry::SpanHolder span(span_name);
+        DB::ProfileEventTimeIncrement<DB::Time::Microseconds> increment(event);
```

Comment:
btw we do have two ways to do the same thing: `ProfileEventTimeIncrement` and  `ProfileEvent::Timer` (looks like `auto timer = DB::CurrentThread::getProfileEvents().timer(event);`). Should we delete one of them? Just contemplating, not asking you to do anything

Reply:
never saw this one. will try to remove it